tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTests.java

```diff
@@ -0,0 +1,259 @@
+    private UserModel userMultiSite;
+    private FileModel sampleContent;
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation()
+    {
+        serverHealth.assertServerIsOnline();
+
+        elasticClient = new RestHighLevelClient(RestClient.builder(new HttpHost("localhost", 9200, "http")));
+        initIndex();
+
+        userSite1 = dataUser.createRandomTestUser();
+        userSite2 = dataUser.createRandomTestUser();
+        userMultiSite = dataUser.createRandomTestUser();
+
+        siteModel = dataSite.usingUser(userSite1).createPrivateRandomSite();
+        SiteModel otherSiteModel = dataSite.usingUser(userSite2).createPrivateRandomSite();
+
+        dataUser.addUserToSite(userSite2, siteModel, UserRole.SiteContributor);
+        dataUser.addUserToSite(userMultiSite, siteModel, UserRole.SiteContributor);
+        dataUser.addUserToSite(userMultiSite, otherSiteModel, UserRole.SiteContributor);
+
+        sampleContent = createContent(FILE_0_NAME, "This is a test", siteModel, userSite1);
+        createContent(FILE_1_NAME, "This is another file", siteModel, userSite1);
+        createContent(FILE_2_NAME, "This is another file", otherSiteModel, userSite2);
+        createContent(FILE_3_NAME, "This is another file", siteModel, userSite2);
+        //remove the user from site, but he keeps ownership on FILE_3_NAME 
+        dataUser.removeUserFromSite(userSite2, siteModel);
+    }
+
+    @TestRail(section = {
+            TestGroup.SEARCH }, executionType = ExecutionType.REGRESSION, description = "Verify that Elasticsearch indexing works as expected.")
+    @Test(groups = { TestGroup.SEARCH })
+    public void fileIndexed() throws Exception
+    {
+        Utility.sleep(1000, 10000, () -> {
+            GetRequest request = new GetRequest(INDEX_NAME);
+            request.id(sampleContent.getNodeRef());
+            GetResponse documentResponse = elasticClient.get(request, RequestOptions.DEFAULT);
+
+            assertTrue(documentResponse.isExists());
+            assertEquals(documentResponse.getSource().get("content"), "This is a test");
+        });
+    }
+
+    @TestRail(section = {
+            TestGroup.SEARCH }, executionType = ExecutionType.REGRESSION, description = "Verify that the simpler Elasticsearch search works as expected.")
+    @Test(groups = { TestGroup.SEARCH })
+    public void searchCanFindAFile() throws Exception
+    {
+        Utility.sleep(1000, 10000, () -> {
+            SearchRequest query = new SearchRequest();
+            RestRequestQueryModel queryReq = new RestRequestQueryModel();
+            queryReq.setQuery("test");
+            query.setQuery(queryReq);
+
+            SearchResponse search = client.authenticateUser(userSite1).withSearchAPI().search(query);
+
+            // this test must found only one documents, while documents in the system are four because 
+            // only one contains the word "test".
+            assertResponseAndResult(search, "test.txt");
```

Comment:
You may use the constant at this point >> FILE_0_NAME

Reply:
Ops, sorry, sometimes IntelliJ is not smart like Eclipse when you "extract" a constant :)